src/ui/movable_rect.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <anton/math/vec2.hpp>
+#include <types.hpp>
+
+namespace nebula {
+  /**
+   * Movable rectangle ( in short MR ) is movable visual copy of MR Button
+   *created
+   *after interaction with it. This structure holds information about objects
+   *current location in the application window and pointer to render function
+   *inherited from MR Button.
+   **/
+  struct Movable_Rect {
+    void (*render_function)(Rect_f32 position);
+    Rect_f32 position;
+
+    /**
+     * Creates a new visual movable object at the middle of the screen following
+     * instructions in passed function.
+     * @param render_function - function used to render object on the screen
+     *                          passed by MR button.
+     * @return pointer to the newly created movable rectangle structure
+     */
+    explicit Movable_Rect(void (*render_function)(Rect_f32 position));
+
+    /**
+     * Renders an object.
+     */
+    void render() const;
+
+    /**
+     * Check whether the object is located at the given mouse location
+     * @param mouse_location - vector, mouse x and y location
+     * @param window_width - GLFWwindow width
+     * @param window_height - GLFWwindow height
+     * @return true if rectangle is under given location
+     */
+    [[nodiscard]] bool is_under_mouse(Vec2 mouse_location, f32 window_width,
```

Comment:
window dimensions could also be a vector

Reply:
Resolved